tests/unit/structures/test_workflow.py

```diff
@@ -781,8 +809,6 @@ def _validate_topology_1(workflow) -> None:
         assert len(workflow.tasks) == 4
         assert workflow.input_task.id == "task1"
         assert workflow.output_task.id == "task4"
-        assert workflow.input_task.id == workflow.tasks[0].id
-        assert workflow.output_task.id == workflow.tasks[-1].id
```

Comment:
Why did you remove these?

Reply:
because this is an incorrect assumption. `input_task` is the first task in the `ordered_tasks` list, not `tasks`. same for `output_task` except the last task.

Reply:
Can you update the test to use `ordered_tasks` then?